ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/ConnectIntegrationTest.java

```diff
@@ -0,0 +1,190 @@
+import kafka.zookeeper.ZooKeeperClientException;
+import org.apache.kafka.connect.json.JsonConverter;
+import org.apache.kafka.connect.storage.StringConverter;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.rules.RuleChain;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Category({IntegrationTest.class})
+public class ConnectIntegrationTest {
+
+  private static final Logger LOG = LoggerFactory.getLogger(ConnectIntegrationTest.class);
+  private static final IntegrationTestHarness TEST_HARNESS = IntegrationTestHarness.build();
+
+  private static final TestKsqlRestApp REST_APP = TestKsqlRestApp
+      .builder(TEST_HARNESS::kafkaBootstrapServers)
+      .withStaticServiceContext(TEST_HARNESS::getServiceContext)
+      .build();
+
+  @ClassRule
+  public static final RuleChain CHAIN = RuleChain
+      .outerRule(Retry.of(3, ZooKeeperClientException.class, 3, TimeUnit.SECONDS))
+      .around(TEST_HARNESS)
+      .around(REST_APP);
+
+  private static ConnectExecutable CONNECT;
+
+  @BeforeClass
+  public static void setUpClass() {
+    CONNECT = new ConnectExecutable(ImmutableMap.<String, String>builder()
+        .put("bootstrap.servers", TEST_HARNESS.kafkaBootstrapServers())
+        .put("group.id", UUID.randomUUID().toString())
+        .put("key.converter", StringConverter.class.getName())
+        .put("value.converter", JsonConverter.class.getName())
+        .put("offset.storage.topic", "connect-offsets")
+        .put("status.storage.topic", "connect-status")
+        .put("config.storage.topic", "connect-config")
+        .put("offset.storage.replication.factor", "1")
+        .put("status.storage.replication.factor", "1")
+        .put("config.storage.replication.factor", "1")
+        .build()
+    );
+    CONNECT.start();
+  }
+
+  @AfterClass
+  public static void tearDownClass() {
+    CONNECT.stop();
+  }
+
+  private KsqlRestClient ksqlRestClient;
+  private Set<String> connectNames;
+
+  @Before
+  public void setupRun() throws Exception {
+    ksqlRestClient = REST_APP.buildKsqlClient(Optional.empty());
+    connectNames = new HashSet<>();
+  }
+
+  @After
+  public void afterRun() {
+    Iterators.consumingIterator(connectNames.iterator())
+        .forEachRemaining(
+            name -> ksqlRestClient.makeKsqlRequest("DROP CONNECTOR " + name + ";"));
+  }
+
+  @Test
+  public void shouldListConnectors() {
+    // Given:
+    create("mock-connector", ImmutableMap.of(
+        "connector.class", "org.apache.kafka.connect.tools.MockSourceConnector"
+    ));
+
+    // When:
+    final RestResponse<KsqlEntityList> response = ksqlRestClient
+        .makeKsqlRequest("SHOW CONNECTORS;");
+
+    // Then:
+    assertThat("expected successful response", response.isSuccessful());
+    assertThat(response.getResponse().size(), is(1));
```

Comment:
Don't feel strongly but we use `hasSize()` in bunch of other tests throughout the repo.